lib/jekyll/configuration.rb

```diff
@@ -175,7 +175,7 @@ def read_config_file(file)
       next_config
     rescue SystemCallError
       if @default_config_file ||= nil
-        Jekyll.logger.warn "Configuration file:", "none"
+        Jekyll.logger.warn "Configuration file:", "none (looking for '#{file}')"
```

Comment:
@parkr and @envygeeks, thanks for the follow-ups. The original case to address here was not about config-file-or-not, it was accidentally running `jekyll` in the `_posts` directory, and getting an error about an illegal URL format (in no specific file), which wasn't helpful.   If Jekyll could have noticed that the call wasn't made from the top of a project *through any means*, that would have been helpful.  It would have also been helpful to tell me *which* file had the illegal URL format in it, had that actually been the problem and not a symptom.